app/services/MatcherPool.scala

```diff
@@ -91,7 +91,7 @@ class MatcherPool(
     */
   def check(query: Check): Future[List[RuleMatch]] = {
     val categoryIds = query.categoryIds match {
-      case None => getCurrentCategories.map { case (_, category, _) => category.id }
+      case None => getCurrentCategories.flatMap { case (_, categories, _) => categories.map(_.id) }
```

Comment:
Can you end up with duplicate category ids here if multiple matchers check for the same category? More interestingly why does getCurrentCategories return something other than categories?

Reply:
Possibly the `CategoryIds` type could be a Set instead of a List.

Reply:
The signature of getCurrentCategories is pretty 'orrid – really it's built to support the rules view. This probably isn't the place to refactor, as it'll mean a rejig of the view, but it sure could use one.

A set rather than a list is sensible.